### PR TITLE
feat: Adding logic  to show shared items that don't have a title

### PR DESCRIPTION
**Pull request by TomDowling**
## Changes
Added logic to set the article page title if the article is shared and has no title.

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1460 #done

### CODE REVIEW

File: packages/webapp/pages/posts/[id]/index.tsx [hunk 1]
```diff
@@ -108,8 +108,16 @@ const PostPage = ({ id, initialData }: Props): ReactElement => {
       sidebarRendered &&
       modalSizeToClassName[ModalSize.Large],
   );
+  const seoTitle = () => {
+    if (
+      post?.type !== PostType.Share ||
+      (post?.type === PostType.Share && post?.title !== null)
+    )
+      return post?.title;
+    return `Shared post at ${post?.source?.name}`;
```
Comment (sshanzel) [inline packages/webapp/pages/posts/[id]/index.tsx:117]: Probably just a personal preference, but I think it is better to isolate the odd one out.

```suggestion
    if (post?.type === PostType.Share && post?.title === null) return `Shared post at ${post?.source?.name}`;
    
    return post?.title;
```

**Review by sshanzel — APPROVED** [review]
Looks amazing! 🚢 

Just had one non-blocking comment.

File: packages/webapp/pages/posts/[id]/index.tsx (hunk 1 above)
Comment (TomDowling) [reply packages/webapp/pages/posts/[id]/index.tsx:117]: I did have it like this initially, but my concern was when there wasn't a title but not Shared, maybe there isn't a use case now but down the line "Shared post at" might not be relevant text to use in these use cases? @sshanzel, however, if you think thats okay i will make this change :)

Comment (rebelchris) [reply packages/webapp/pages/posts/[id]/index.tsx:117]: I think Lee's suggestion is ok, anything non squads should have mandatory title these days for it to become visible.
Let's go with this, if it becomes an issue we can always address it again 🙏

**Review by rebelchris — APPROVED** [review]

**Pull request merged**